src/Perpetuum.Bootstrapper/PerpetuumBootstrapper.cs

```diff
@@ -2421,6 +2421,23 @@ private void RegisterRifts()
             {
                 return zone =>
                 {
+
+                    var zoneConfigs = Db.Query().CommandText("SELECT maxrifts FROM zoneriftsconfig WHERE zoneid = @zoneId")
+                    .SetParameter("@zoneId", zone.Id)
+                    .Execute();
+                    if (zoneConfigs.Count < 1) 
+                    {
+                        return null;
+                    }
+
+                    var record = zoneConfigs[0];
+                    var maxrifts = record.GetValue<int>("maxrifts");
+
+                    if (maxrifts < 1) 
+                    {
+                        return null;
+                    }
+
```

Comment:
This might be a bit overkill.  Really this is registering a function (only used in the other function about to be registered below) to pick a SpawnPosFinder based on if the zone is Gamma or non-gamma (the terraform check)

This code is invoked only on line 2493, which is after the check (same DB lookup) if the zone has any random rifts to spawn.
You can safely remove this code, it saves roundtrips to the DB on boot, and is more DRY.  Otherwise Id advocate for this logic to be better encapsulated/formalized somewhere instead of this raw query.

Reply:
Could almost be persuaded to inline this logic into the func below.
**Unless** it is being called elsewhere in the bootstrapper, which is not always immediately obvious..